src/PrincipalToken.sol

```diff
@@ -1,86 +1,82 @@
 // SPDX-License-Identifier: CC0-1.0
-pragma solidity ^0.8.20;
+pragma solidity 0.8.20;
 
-import {IDelegateToken, TokenType} from "./interfaces/IDelegateToken.sol";
+import {IDelegateToken, IDelegateRegistry} from "./interfaces/IDelegateToken.sol";
 
-import {LibString} from "solady/utils/LibString.sol";
-import {Base64} from "solady/utils/Base64.sol";
+import {Strings} from "openzeppelin/utils/Strings.sol";
```

Comment:
OpenZeppelin has worse gas performance than Solady, somewhat inclined to keep Solady's functions over OZ

Reply:
I think `Strings` and `Base64` are only used in `tokenURI` so that shouldn't affect regular gas use.